src/mono/mono.proj

```diff
@@ -581,6 +581,8 @@
     </ItemGroup>
     <ItemGroup>
       <_MonoCMakeArgs Include="-DMONO_COMPONENTS_RID=$(TargetOS)-$(TargetArchitecture)" />
+      <_MonoCMakeArgs Condition="'$(TargetArchitecture)' == 'wasm'" Include="-DCLR_CMAKE_HOST_ARCH=$(BuildArchitecture)" />
+      <_MonoCMakeArgs Condition="'$(TargetArchitecture)' != 'wasm'" Include="-DCLR_CMAKE_HOST_ARCH=$(TargetArchitecture)" />
```

Comment:
This is copied from couple of lines above, from current `main`: https://github.com/dotnet/runtime/blob/063adfe9f2925c5c6c74e5d30d04a7d5a085e144/src/mono/mono.proj#L258-L259

Reply:
Added a comment. It is the case until we completely switch mono to use the common infra.

Reply:
Thanks, but it is still not really clear to me why this is different on wasm vs. non-wasm?

Reply:
I copied this existing condition from this same file. There is no comment on line 769 either. Without this, wasm builds fail. So the comment I've added is based on my understanding of the situation. I'm sure there is a better way to elaborate it, but depending on the audience, I guess it will still fall short until we explain everything. 😅

Wasm is always cross-compile meaning `host` is always different than `target`. But when we add `build` dimension; i.e. we are in our _other_ cross-compiling environment (e.g. `inux-x64` (the system running `build.sh`) -> `linux-arm64` (for which the cross/aot tools are compiled in stage0) -> wasm (the target emitted by those cross/aot tools in stage 1)) then `host` != `target` != `build` holds and it fails wasm builds without this flip.

In coreclr, tools like crossgen2 are built for multi-tagets, so we can do things like `crossgen2 --targetarch arm64` on x64 machine. That cuts off the requirement of dealing with the third dimension. I suppose mono cross/aot tools are not on such multi-target plan.